sympy/functions/combinatorial/numbers.py

```diff
@@ -2070,6 +2075,11 @@ def find_motzkin_numbers_in_range(x, y):
 
     @staticmethod
     def find_first_n_motzkin(n):
+        try:
+            n = as_int(n)
+        except:
+            raise ValueError('The provided number must be a positive integer')
+            return False
```

Comment:
Here it indeed makes sense to raise an exception, but no need for the return as that is dead code.

Reply:
Looking at the test error, you will still need to check that the integer is positive. If not, raise the same exception.